Gems/SurfaceData/Code/Source/SurfaceDataSystemComponent.cpp

```diff
@@ -59,7 +59,14 @@ namespace SurfaceData
                 ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Common)
                 ->Attribute(AZ::Script::Attributes::Category, "Vegetation")
                 ->Attribute(AZ::Script::Attributes::Module, "surface_data")
-                ->Event("GetSurfacePoints", &SurfaceDataSystemRequestBus::Events::GetSurfacePoints)
+                ->Event(
+                    "GetSurfacePoints",
+                    [](SurfaceData::SurfaceDataSystem* handler, const AZ::Vector3& inPosition, const SurfaceTagVector& desiredTags) -> SurfaceData::SurfacePointList
+                    {
+                        SurfaceData::SurfacePointList result;
+                        handler->GetSurfacePoints(inPosition, desiredTags, result);
+                        return result;
+                    })
```

Comment:
Actually, thinking about this further, while this code is exactly correct for turning the list into an output, the SurfacePointList structure is going to be a difficult one to expose and work with from script.

As an alternative, you could change this to return a vector, and add some code after the GetSurfacePoints() call to call result.EnumeratePoints to add all the results to the vector. Since only one input position is passed in, you don't need to worry about which output points map to which input points. (This is a trickier problem when getting the surface points from a region)

The code would look _roughly_ like this, I think:
```
                    [](SurfaceData::SurfaceDataSystem* handler, const AZ::Vector3& inPosition, const SurfaceTagVector& desiredTags) -> AZStd::vector<SurfacePoint>
                    {
                        AZStd::vector<SurfacePoint> result; 
                        SurfaceData::SurfacePointList surfacePoints;
                        handler->GetSurfacePoints(inPosition, desiredTags, surfacePoints);
points.EnumeratePoints(
                [this, &result](
                    [[maybe_unused]] size_t inPositionIndex, const AZ::Vector3& position, const AZ::Vector3& normal,
                    const SurfaceData::SurfaceTagWeights& masks) -> bool
                {
                    result.emplace_back(position, normal, masks);
                    return true;
                });
                        return result;
                    })

```